include/nbl/video/debug/debug.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __NBL_DEBUG_H_INCLUDED__
+#define __NBL_DEBUG_H_INCLUDED__
```

Comment:
rename header to `SDebugCallback`

and write a big comment/documentation that the callback function can and will be called from multiple threads (possibly even simultaneously) so user needs to account for that.